Clarify list-of-values deprecation for non-list search attribute types (#4593)
temporalio/documentation: docs/encyclopedia/visibility/search-attributes.mdx

  For example, if the key `ProductId` has the value of `2dd29ab7-2dd8-4668-83e0-89cae261cfb1`:
  - As a **Keyword** it would be matched only by `ProductId = "2dd29ab7-2dd8-4668-83e0-89cae261cfb1`.
  - As a **Text** it would be matched by `ProductId = 2dd8`, which could cause unwanted matches.
- With Temporal Server v1.20 and later, the **Keyword** type supports only a single value.
  To store a list of values, use **KeywordList**.

:::warning List-of-values deprecation for non-list types

> Storing a list of values in a non-list Search Attribute type (for example, multiple integers in an **Int** attribute or multiple strings in a **Keyword** attribute) is deprecated and will be enforced after September 30, 2026.

Reviewer: ```suggestion
Storing a list of values in a non-list Search Attribute type (such as multiple strings in a **Keyword** attribute) is deprecated.  
New namespaces on Temporal Cloud are already blocked from using this pattern. Existing Namespaces that rely on it will need to migrate before Sep. 30, 2026. After that date, Workflow Executions with invalid Search Attribute values will fail. 
```

Moving the consequence up and shortening the sentences in brackets.

Author: The original line seemed better to me bc it called out Ints specifically. multiple integers in an **Int** attribute or multiple strings in a **Keyword** attribute

Reviewer: This is a text flow issue. It's generally not great to put long sentences in brackets as it disrupts reader attention. 

We had two examples here before. For us who are looking closely at this content, this looks good, but for most users who just want to skim this already distracting admonition, especially if you consider we are going to have a lot of users on this page who are just learning what CSAs are and probably don't care about this deprecation at all, this doesn't work well. 

We already provide remediation strategies that cover both of these cases later on, so I think it's okay to keep the drive-by mention here short. If you feel strongly that we should keep both examples, we can, but my suggestion here is to keep it short. 

Also, if the incorrect `ints` list occur more frequently and the incorrect keywords list, we can change to that here.

This pattern was unintended behavior inherited from the underlying Elasticsearch encoding, not a supported feature.

If you are currently storing lists in **Keyword** attributes, migrate to **KeywordList**.
If you are storing lists in **Int** attributes, consider either migrating to **KeywordList** (storing integers as strings, with string-based comparison only) or splitting into multiple individual **Int** attributes.

> Temporal Cloud is actively enforcing this restriction.

Reviewer: ```suggestion
```

> New namespaces are already blocked from using this pattern, and existing namespaces that rely on it will need to migrate.

Reviewer: Not necessarily opposed to the red admonition, but we should spell out the consequences if they don't migrate if it's destructive

> New namespaces are already blocked from using this pattern, and existing namespaces that rely on it will need to migrate.

Reviewer: ```suggestion
```
